File: src/CSharp/IO/WriteTIFF.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Windows.Forms;
using System.Diagnostics;
using System.IO;

using Kitware.VTK;

namespace ActiViz.Examples {
   public partial class Form1 : Form {
      public Form1() {
         InitializeComponent();
      }


      private void renderWindowControl1_Load(object sender, EventArgs e) {
         try {
            WriteTIFF();
         }
         catch(Exception ex) {
            MessageBox.Show(ex.Message, "Exception", MessageBoxButtons.OK);
         }
      }


      private void WriteTIFF() {
         // Path to vtk data must be set as an environment variable
         // VTK_DATA_ROOT = "C:\VTK\vtkdata-5.8.0"
         vtkTesting test = vtkTesting.New();
         string root = test.GetDataRoot();
         string filePath = System.IO.Path.Combine(root, @"Data\test_tiff.tif");

         vtkImageCanvasSource2D imageSource = vtkImageCanvasSource2D.New();
         imageSource.SetScalarTypeToUnsignedChar();
         imageSource.SetExtent(0, 9, 0, 9, 0, 0);
         imageSource.SetNumberOfScalarComponents(3);
         imageSource.SetDrawColor(0, 0, 0, 0);
         imageSource.FillBox(0, 9, 0, 9);
         imageSource.SetDrawColor(255, 0, 0, 0);
         imageSource.FillBox(5, 7, 5, 7);
         imageSource.Update();

         vtkTIFFWriter tiffWriter = vtkTIFFWriter.New();
         tiffWriter.SetFileName(filePath);
         tiffWriter.SetInputConnection(imageSource.GetOutputPort());
         tiffWriter.Write();

         // Read and display file for verification that it was written correctly
         vtkTIFFReader reader = vtkTIFFReader.New();
         if(reader.CanReadFile(filePath) == 0) {
            MessageBox.Show("Cannot read file \"" + filePath + "\"", "Error", MessageBoxButtons.OK);
            return;
         }
         reader.SetFileName(filePath);
         reader.Update();

         // Visualize
         vtkImageViewer2 imageViewer = vtkImageViewer2.New();
         imageViewer.SetInputConnection(reader.GetOutputPort());
         // get a reference to the renderwindow of our renderWindowControl1
         vtkRenderWindow renderWindow = renderWindowControl1.RenderWindow;
         // renderer
         vtkRenderer renderer = renderWindow.GetRenderers().GetFirstRenderer();
         // set background color
         renderer.SetBackground(0.2, 0.3, 0.4);
         imageViewer.SetRenderer(renderer);
         renderer.ResetCamera();
      }
   }
}

File: src/Java/IO/WriteTIFF.java
import vtk.vtkNativeLibrary;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;
import vtk.vtkImageCanvasSource2D;
import vtk.vtkTIFFWriter;
import vtk.vtkTIFFReader;
import vtk.vtkImageActor;
import vtk.vtkInteractorStyleImage;

public class WriteTIFF
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------
  
  public static void main(String args[]) 
  {

    //parse command line arguments
    if (args.length != 1) 
    {
      System.err.println("Usage: java -classpath ... Filename(.tif) e.g OutputFile.tif");
      return;
    }
    String Filename = args[0];
     
    vtkImageCanvasSource2D imageSource = new vtkImageCanvasSource2D();
    imageSource.SetScalarTypeToUnsignedChar();
    imageSource.SetExtent(0,9,0,9,0,0);
    imageSource.SetNumberOfScalarComponents(3);
    imageSource.SetDrawColor(0, 0, 0, 0);
    imageSource.FillBox(0,9,0,9);
    imageSource.SetDrawColor(255, 0, 0, 0);
    imageSource.FillBox(5,7,5,7);
    imageSource.Update();
    
    vtkTIFFWriter tiffWriter = new vtkTIFFWriter();
    tiffWriter.SetFileName(Filename);
    tiffWriter.SetInputConnection(imageSource.GetOutputPort());
    tiffWriter.Write();
    
    //Read and display for verification
    vtkTIFFReader reader = new vtkTIFFReader();
    reader.SetFileName(Filename);
    reader.Update();
    
    vtkImageActor actor = new vtkImageActor();
    actor.GetMapper().SetInputConnection(reader.GetOutputPort());

    //Create the renderer, render window and interactor.
    vtkRenderer ren = new vtkRenderer();
    vtkRenderWindow renWin = new vtkRenderWindow();
    renWin.AddRenderer(ren);
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);
    
    vtkInteractorStyleImage style = new vtkInteractorStyleImage();
    iren.SetInteractorStyle(style);

    ren.AddActor(actor);

    renWin.SetSize(300,300);
    renWin.Render();

    iren.Initialize();
    iren.Start();

  }
}
